File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        unordered_map<int,int>mp ;
        unordered_set<int>st ; 
        int n = nums.size() ;
        for( int x : nums ){
            mp[x]++; 
        } 
        for( int i = 0 ; i < n ; i++ ){
            if( mp[nums[i]] >= 2 ){
                st.insert(nums[i]) ; 
            }
        }
        vector<int>result(st.begin(),st.end()) ;

        return result ; 
    }
};

File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        ArrayList<Integer>arr = new ArrayList<>() ; 
        HashMap<Integer,Integer>mp = new HashMap<>() ; 
        for( int x : nums){
           mp.put(x,mp.getOrDefault(x,0)+1) ; 
        }
        for( int key : mp.keySet()){
            if(mp.get(key) > 1 ){
                arr.add(key) ; 
            }
        }
        return arr ;
    }
}